lightningd/channel.c

```diff
@@ -555,11 +609,18 @@ struct channel *new_channel(struct peer *peer, u64 dbid,
 	channel->scid = tal_steal(channel, scid);
 	channel->old_scids = tal_dup_talarr(channel, struct short_channel_id, old_scids);
 	channel->alias[LOCAL] = tal_dup_or_null(channel, struct short_channel_id, alias_local);
+	/* All these possible short_channel_id variants go in the lookup table! */
+	if (channel->scid)
+		chanmap_add(peer->ld, channel, *channel->scid);
+	if (channel->alias[LOCAL])
+		chanmap_add(peer->ld, channel, *channel->alias[LOCAL]);
```

Comment:
In opening_control.c's `stub_chan()` we set `scid`, `alias[LOCAL]`, and `alias[REMOTE]` to `1:1:1` which I believe causes `chanmap_add` to crash on the duplicate entry.

We either need to support duplicates in `chanmap` or tweak `stub_chan()` to use different values for the `alias` and `scid`.

A related question is if we'll ever have two stub channels at once? I'd imagine not but I'm not sure.

An explicit error message on duplicate scid/alias values could be helpful here to make any future mistakes around this easier to catch.

Reply:
Another idea is maybe we make `1:1:1` special and never add it to the `chanmap` as it's not a valid scid anyway.

Reply:
Ah, I did fix.  Didn't push!  Done.